arch/arm/src/nrf91/Kconfig

```diff
@@ -742,4 +742,22 @@ config NRF91_MODEM_AT
 	bool "Modem AT interface support"
 	default y
 
+config NRF91_MODEM_LTEM
+	bool "Modem LTE-M mode support"
+	default y
+
+config NRF91_MODEM_NBIOT
+	bool "Modem NB-IoT mode support"
+	default n
+
+config NRF91_MODEM_GNSS
+	bool "Modem GNSS support"
+	depends on SENSORS_GPS
+	default y
+
+config NRF91_MODEM_PREFERENCE
+	int "Modem mode preference"
+	default 0
+	range 0 4
```

Comment:
@raiden00pl is there some documentation/explanation about these 5 preference types?

Reply:
I added help message and a note about the doc where we can find some details